examples/cat_facts/web-nextjs/src/app/page.tsx

```diff
@@ -110,37 +43,37 @@ const Home: NextPage = () => {
 
       <main>
         <section className="section title has-text-centered">
-          <p>{state.platform}</p>
+          <p>{view.platform}</p>
         </section>
         <section className="section container has-text-centered">
-          {state.image && (
+          {view.image && (
             // eslint-disable-next-line @next/next/no-img-element
             <img
               alt="A funny cat. Or at least a cute one."
-              src={state.image?.href}
+              src={view.image?.href}
               style={{ height: "200px" }}
             />
           )}
         </section>
         <section className="section container has-text-centered">
-          <p>{state.fact}</p>
+          <p>{view.fact}</p>
         </section>
         <div className="buttons container is-centered">
           <button
             className="button is-primary is-danger"
-            onClick={() => dispatch(new EventVariantClear())}
+            onClick={() => update(new EventVariantClear(), setView)}
```

Comment:
passing `setView` into these looks really strange to me, and easy to miss. Maybe we could add a level of indirection that links up the callback with the core wrapper?

Reply:
To me it reads as "update the core with this event and then call setView with any resulting view model changes". It's not so bad. 😓

Reply:
Hmm, don't love it, but I'm okay with it.